docs/input-snmptrap.asciidoc

```diff
@@ -27,22 +27,34 @@ Read SNMP trap messages as events
 
 Resulting `message` field resembles:
 [source,ruby]
-  #<SNMP::SNMPv1_Trap:0x6f1a7a4 @varbind_list=[#<SNMP::VarBind:0x2d7bcd8f @value="teststring",
-  @name=[1.11.12.13.14.15]>], @timestamp=#<SNMP::TimeTicks:0x1af47e9d @value=55>, @generic_trap=6,
-  @enterprise=[1.2.3.4.5.6], @source_ip="127.0.0.1", @agent_addr=#<SNMP::IpAddress:0x29a4833e @value="\xC0\xC1\xC2\xC3">,
-  @specific_trap=99>
+{"agent_addr"=>"192.168.1.40", "generic_trap"=>6, "specific_trap"=>15511, "enterprise"=>"1.3.6.1.2.1.1.1", "variable_bindings"=>{"1.3.6.1.2.1.1.2.0"=>"test one", "1.3.6.1.2.1.1.1.0"=>"test two"}, "type"=>"V1TRAP", "community"=>"public", "version"=>1, "timestamp"=>1500}
 
 
 [id="plugins-{type}s-{plugin}-ecs"]
-==== Compatibility with the Elastic Common Schema (ECS)
+==== Event Metadata and the Elastic Common Schema (ECS)
 
-Because SNMP data has specific field names based on OIDs, we recommend setting a <<plugins-{type}s-{plugin}-target>>.
-The source host field changes based on <<plugins-{type}s-{plugin}-ecs_compatibility>>.
+When ECS <<plugins-{type}s-{plugin}-ecs_compatibility>> is enabled, this plugin adds SNMP trap metadata to each event. The value is stored in the @metadata where it can be used by other plugins in your pipeline.
```

Comment:
Hey @jsvd! Thanks for reviewing!

I agree with you on always adding the `@metadata` fields regardless of ECS status, but I'm not sure about adding all those metadata fields on the event by default when ECS is enabled. The data those PDUs carry (variable bindings) are already being added as fields, and maybe, not all the extra data are interesting to users. 

When I added those `@metadata` fields, I was thinking on providing PDU's details to users so they can use it on the pipeline to filter, group, etc. I haven't though about changing the existing event data. WDYT? If that might be interesting to users, I'd be okay adding them :)

Reply:
I don't think we'll be able to figure out what ECS means for this integration during this review. We can still make ECS compatibility a requirement for GA, but shouldn't block this PR from being merged.
My suggestion is to prevent the plugin from running if ECS enabled right now, where with ECS disabled it just adds that data to `@metadata`.

Reply:
I think preventing the plugin from running wouldn't be backward compatibility with the existing traps plugin. It currently support both ECS disabled and enabled, mapping the `host` or `[host][ip]` field respectively.

Reply:
I've copied the `@metadata` field names pattern from other plugins (ECS enabled names), such as beats and http. Do you think mapping those fields as it's for both, ECS enabled and disabled, wouldn't be a good idea? considering the metadata fields are new and wouldn't break existing behaviour?

Reply:
For now, we've agreed on adding the `@metadata` for both, ECS enabled and disabled.